GUI/Utils/DarkModeForms/DarkModeCS.cs

```diff
@@ -0,0 +1,1037 @@
+            /// </summary>
+            DWMWA_CAPTION_COLOR,
+
+            /// <summary>
+            /// Use with DwmSetWindowAttribute. Specifies the color of the caption text. The pvAttribute parameter points to a value of type COLORREF. This value is supported starting with Windows 11 Build 22000.
+            /// </summary>
+            DWMWA_TEXT_COLOR,
+
+            /// <summary>
+            /// Use with DwmGetWindowAttribute. Retrieves the width of the outer border that the DWM would draw around this window. The value can vary depending on the DPI of the window. The pvAttribute parameter points to a value of type UINT. This value is supported starting with Windows 11 Build 22000.
+            /// </summary>
+            DWMWA_VISIBLE_FRAME_BORDER_THICKNESS,
+
+            /// <summary>
+            /// The maximum recognized DWMWINDOWATTRIBUTE value, used for validation purposes.
+            /// </summary>
+            DWMWA_LAST,
+        }
+
+        [Flags]
+        public enum DWM_WINDOW_CORNER_PREFERENCE
+        {
+            DWMWCP_DEFAULT = 0,
+            DWMWCP_DONOTROUND = 1,
+            DWMWCP_ROUND = 2,
+            DWMWCP_ROUNDSMALL = 3
+        }
+
+        public const int EM_SETCUEBANNER = 5377;
+
+        public static IntPtr GetHeaderControl(ListView list)
+        {
+            const int LVM_GETHEADER = 0x1000 + 31;
+            return SendMessage(list.Handle, LVM_GETHEADER, IntPtr.Zero, "");
+        }
+
+        #endregion
+
+        /// <summary>'true' if Dark Mode Color is set in Windows's Settings.</summary>
+        public bool IsDarkMode { get; set; }
+
+        /// <summary>Windows Colors. Can be customized.</summary>
+        public ThemeColors ThemeColors { get; set; }
+
+        /// <summary>Constructor.</summary>
+        public DarkModeCS(bool debugTheme = false)
+        {
+            DebugTheme = debugTheme;
+            SystemEvents.UserPreferenceChanged += new UserPreferenceChangedEventHandler(OnUserPreferenceChanged);
+        }
+
+        #region Public Methods
+
+        /// <summary>This tries to style and automatically apply Windows Dark Mode (if enabled) to a Form.</summary>
+        /// <param name="_Form">The Form to become Dark</param>
+        public void Style(Form _Form)
+        {
+            ApplyTheme(_Form);
+        }
+
+        /// <summary>Recursively apply the Colors from 'ThemeColors' to the Control and all its childs.</summary>
+        /// <param name="control">Can be a Form or any Winforms Control.</param>
+        public void ThemeControl(Control control)
+        {
+            BorderStyle BStyle = BorderStyle.FixedSingle;
+            FlatStyle FStyle = FlatStyle.Flat;
+
+            //Change the Colors only if its the default ones, this allows the user to set own colors:
+            if (control.BackColor == SystemColors.Control || control.BackColor == SystemColors.Window)
```

Comment:
Change this to pattern mathcing because BackColor getter does a bunch of stuff, no reason to call it twice